en/components/linear_progress.md

```diff
@@ -9,29 +9,267 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
+```typescript
+// app.module.ts
+
+...
+import { IgxProgressBarModule } from 'igniteui-angular/main';
+
+@NgModule({
+    ...
+    imports: [..., IgxProgressBarModule],
+    ...
+})
+export class AppModule {}
+```
+
+And now let's create a simple example in which we are going to simulate a continuous process, that is triggered on button click. In addition you can notice
+that the speed of loading depends on the `max` attribute that we are set to our `igx-linear-bar`. In first case our max value is set to 200 and loads quite slower,
+than in the second where is set to 100(default value). So let's see the code...
+
 ```html
-<igx-linear-bar [striped]="false" [value]="currentValue" [max]="200">
-</igx-linear-bar>
+...
+<section class="sample-content">
+    <h4 class="sample-title">Linear progress bar</h4>
+    <div class="linear-container">
+        <h5>Max value: 200</h5>
+        <igx-linear-bar [striped]="false" [max]="200" [value]="0"></igx-linear-bar>
+    </div>
+    <div class="linear-container">
+        <h5>Max value: 100</h5>
+        <igx-linear-bar [striped]="false" [max]="100" [value]="0" type="danger" ></igx-linear-bar>
+    </div>
+</section>
+....
+<div class="button-container">
+    <p>Press the button to start updating the bars</p>
+    <button igxButton="fab" igxButtonBackground="#333" igxRipple="white" (click)="tick()">
+        <igx-icon fontSet="material" [name]="changeIcon()"></igx-icon>
+    </button>
+</div>
+...
 ```
+```typescript
+  @ViewChildren(IgxLinearProgressBarComponent, { read: IgxLinearProgressBarComponent })
+  public linearBars: QueryList<IgxLinearProgressBarComponent>;
+
+  public interval: any;
+
+  public updateValue() {
+      this.linearBars.map((bar) => bar.value += this.randomIntFromInterval(1, 3));
+      const shouldStop = this.linearBars.toArray().every((bar) => bar.value >= bar.max);
+      if (shouldStop) {
+        this.interval = clearInterval(this.interval);
+      }
+  }
+
+  public tick() {
+      if (this.interval) {
+          this.interval = clearInterval(this.interval);
+          return;
+      }
+      this.interval = setInterval(this.updateValue.bind(this), 60);
+  }
+
+  private randomIntFromInterval(min: number, max: number) {
+      return Math.floor(Math.random() * (max - min + 1) + min);
+  }
+```
+If all went well, you should see something like the following in your browser:
+<div class="sample-container loading" style="height:500px">
+    <iframe id="linear-progressbar-iframe" frameborder="0" seamless width="100%" height="100%" src="{environment:demosBaseUrl}/linear-progressbar" onload="onSampleIframeContentLoaded(this);"></iframe>
+</div>
+<div>
+<button data-localize="stackblitz" class="stackblitz-btn" data-iframe-id="linear-progressbar-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+<div class="divider--half"></div>
+And now let's enhance our example and create a different types of loading bars, that can be striped or not.
```

Comment:
and create different types (if more than one type)
and create a different type (if only one type)